include/desktop/toplevel.h

```diff
@@ -0,0 +1,44 @@
+/* Copyright (c), Niclas Meyer <niclas@countingsort.com>
+ *
+ * This Source Code Form is subject to the terms of the Mozilla Public
+ * License, v. 2.0. If a copy of the MPL was not distributed with this file,
+ * You can obtain one at https://mozilla.org/MPL/2.0/.
+ */
+
+#ifndef KIWMI_DESKTOP_TOPLEVEL_H
+#define KIWMI_DESKTOP_TOPLEVEL_H
+
+struct wlr_surface;
+struct kiwmi_desktop;
+
+enum kiwmi_toplevel_type {
+    KIWMI_TOPLEVEL_VIEW,
+    KIWMI_TOPLEVEL_LAYER,
+};
+
+struct kiwmi_toplevel {
```

Comment:
Apparently, *toplevel* is more widely used to refer only to xdg-shell or xwayland, but not layer-shell. Should think of a name that doesn’t conflict with other compositors’ terminology. (Perhaps i have accidentally swapped *toplevel* and *view*? 🤷)

Reply:
Toplevel is the term used for windows, layershell represents viewports of surfaces.

Reply:
Self-note in case me forgets before me gets to renaming it: weston uses `desktop_surface` (pointed out to me by ammen99)